File: client/src/app/core/model/Review.ts
import { Product } from './Products';
import { User } from './User';

export interface Review {
  reviewId: Number;
  productId: Number;
  userId: Number;
  status: String;
  rating: String;
  content: String;
}

export interface ReviewResponse {
  review: Review;
  user: User;
  product: Product;
}

export interface ReviewDetail {
  reviewId: Number;
  userName: String;
  productName: String;
  status: String;
  rating: Number;
  content: String;
}

File: server/src/main/java/com/store/product_application/model/Review.java
package com.store.product_application.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.store.product_application.common.enums.ReviewStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/*
 * @author Pukhraj Singh
 * @version 1.0
 */
@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "reviews")
public class Review {

    @Id
    @Column(name = "review_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long reviewId;

    private String content;

    @Enumerated(EnumType.STRING)
    private ReviewStatus status;

    private long userId;
    private long productId;

    private int rating;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Date createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Date updatedAt;
}
